src/classes/modules/java.base/jdk/internal/misc/SharedSecrets.java

```diff
@@ -77,6 +78,15 @@ public static JavaNetAccess getJavaNetAccess() {
     return javaNetAccess;
   }
 
+  public static void setJavaNetURLAccess(JavaNetURLAccess jnua) {
+    javaNetURLAccess = jnua;
+  }
+
+  public static JavaNetURLAccess getJavaNetURLAccess() {
+    if (javaNetURLAccess == null)
+      unsafe.ensureClassInitialized(java.net.URL.class);
+    return javaNetURLAccess;
```

Comment:
Is this method allowed to return null, or should it also initialize an instance?